src/Storage/WebStorage.php

```diff
@@ -4,40 +4,28 @@
 
 namespace SpiralPackages\Profiler\Storage;
 
-use DateTimeInterface;
 use Symfony\Contracts\HttpClient\HttpClientInterface;
 use SpiralPackages\Profiler\Converter\ConverterInterface;
 use SpiralPackages\Profiler\Converter\NullConverter;
 
 final class WebStorage implements StorageInterface
 {
-    private HttpClientInterface $httpClient;
-    private string $endpoint;
-    private string $method = 'POST';
-    private array $options = [];
-    private ConverterInterface $converter;
-
     public function __construct(
-        HttpClientInterface $httpClient,
-        string $endpoint,
-        string $method = 'POST',
-        array $options = [],
-        ?ConverterInterface $converter
+        private readonly HttpClientInterface $httpClient,
+        private readonly string $endpoint,
+        private readonly string $method = 'POST',
+        private array $options = [],
+        private readonly ConverterInterface $converter = new NullConverter(),
```

Comment:
If we use the property promotion, we can change the php value in composer from 7.4 to 8.1.